File: math/number-theory/zeno.java
/**
 * Copyright https://github.com/microwind
 * Zeno's Paradox - Dichotomy Paradox
 */

class Zeno {
    public static void zenoParadox(double totalDistance, int steps) {
        double remaining = totalDistance;
        System.out.println("Zeno's Dichotomy Paradox:");
        for (int i = 1; i <= steps; i++) {
            remaining /= 2;
            System.out.printf("Step %d: remaining distance = %.6f%n", i, remaining);
        }
    }
    
    public static void main(String[] args) {
        zenoParadox(1.0, 10);
    }
}

File: math/number-theory/zeno.rs
// Copyright https://github.com/microwind
// Zeno's Paradox - Dichotomy Paradox

fn zeno_paradox(total_distance: f64, steps: i32) {
    let mut remaining = total_distance;
    println!("Zeno's Dichotomy Paradox:");
    for i in 1..=steps {
        remaining /= 2.0;
        println!("Step {}: remaining distance = {:.6}", i, remaining);
    }
}

fn main() {
    zeno_paradox(1.0, 10);
}
